src/primitives/transaction.h

```diff
@@ -43,7 +44,10 @@ class COutPoint
 
     friend bool operator<(const COutPoint& a, const COutPoint& b)
     {
-        return std::tie(a.hash, a.n) < std::tie(b.hash, b.n);
+        DataStream ser_a, ser_b;
+        ser_a << a;
+        ser_b << b;
+        return Span{ser_a} < Span{ser_b};
```

Comment:
I fail to see how it's not functionally equivalent?

Reply:
Although, to your point, `a.hash < b.hash` is doing a lexicographic comparison:

https://github.com/bitcoin/bitcoin/blob/e623d8a05ad4e2b81e19c4f10b22e2ce4db35b8d/src/uint256.h#L54

So maybe it would be more efficient to just serialize `a.vout, b.vout` as little-endian here, rather than serializing the whole `COutPoint`?

Reply:
But 1 sorts before 256 lexicographically? So it's lexicographical *serialized* order, i see.

Reply:
Yep, good point. I added a comment to clarify that this is a lexicographic sort on the serialized outpoint. I think this is preferable to what we were doing before since its based on the protocol definition of an outpoint: `<32-byte txid, little-endian>:<4-byte vout, little-endian>`.